src/ethereum/utils.py

```diff
@@ -59,3 +76,354 @@ def ceil32(value: Uint) -> Uint:
         return value
     else:
         return value + ceiling - remainder
+
+
+def to_valid_address(data: bytes) -> Address:
+    """
+    Convert a byte stream to a valid address (20 bytes).
+
+    Parameters
+    ----------
+    data :
+        The string to be converted to bytes.
+
+    Returns
+    -------
+    byte_stream : `bytes`
+        The byte stream obtained from the passed string.
+    """
+    if len(data) == 20:
+        return data
+
+    if len(data) < 20:
+        return data.rjust(20, b"\x00")
+
+    return data[-20:]
+
+
+def to_bytes(data: str) -> Bytes:
```

Comment:
This function is probably a bad idea.  We had it in `eth-utils` a long time back and ultimately kept running up against the fact that not all `str` types can be naively decoded into a `bytes` type.

Reply:
I agree Piper. Currently it's being used only in the tests module, so maybe I'll add a note mentioning that this can't work against all input formats?

Reply:
I'd actually just move it to the tests module.

Reply:
We dealt with this via the following code organization convention.

The module `library.tools` is where things go that are **not** allowed to be used by the rest of the library.  They are tools built on-top of or alongside the library but are not part of the core functionality.  This keeps them in the main library namespace, allowing them to be linted and type checked, but avoiding the problem of mixing in things like testing utilities with the rest of the library code.

Putting stuff in `./tests/...` that you cross import proved problematic in some situations since `./tests/` tends to not be a valid python module (since it doesn't and shouldn't have an `__init__.py` ).